plugins/modules/one_host.py

```diff
@@ -152,16 +152,18 @@ def __init__(self):
     def allocate_host(self):
         """
         Creates a host entry in OpenNebula
+        self.one.host.allocate returns ID of a host
         Returns: True on success, fails otherwise.
 
         """
-        if not self.one.host.allocate(self.get_parameter('name'),
-                                      self.get_parameter('vmm_mad_name'),
-                                      self.get_parameter('im_mad_name'),
-                                      self.get_parameter('cluster_id')):
-            self.fail(msg="could not allocate host")
-        else:
+        allocate_results = self.one.host.allocate(self.get_parameter('name'),
+                                                  self.get_parameter('vmm_mad_name'),
+                                                  self.get_parameter('im_mad_name'),
+                                                  self.get_parameter('cluster_id'))
+        if allocate_results or allocate_results == 0:
```

Comment:
What does `allocate()` returns when it fails? `None`?

Reply:
nah, it returns something like this, but it seems like a module firstly checks for existence and only then tries to allocate host, might as well remove this statement or make `try-except` 
![image](https://github.com/user-attachments/assets/57d59cc7-70aa-490d-b2e5-fa13592822c7)

Reply:
well, this `if` block exists to check if the allocation was successful or not and, from what you wrote, it looks like the return value for that method is not the right way to test that - if it raises an exception upon failure, that's what should be tested (with the `try-except`). My $0.02

Reply:
yeah, checking for `rc=0` seems pointless as it will fail immediately :)
changed in a last commit